File: Python-Test/test_mysdk.py
# coding: utf-8


import sys
import unittest

if sys.version_info.major == 3:
    from xmlrpc.client import ServerProxy
else:
    from xmlrpclib import ServerProxy


RPC_SERVER_URI = 'http://192.168.1.4:8000'

class MySdkTests(unittest.TestCase):

    def setUp(self):
        self.sdk = ServerProxy(RPC_SERVER_URI).sdk

    def tearDown(self):
        pass

    def test_add(self, a=123, b=456):
        result = self.sdk.add(a, b)
        print('\nsdk.add(%d, %d) = %d' % (a, b, result))
        self.assertEqual(result, 579)

    def test_subtract(self, a=2015, b=1999):
        result = self.sdk.subtract(a, b)
        print('\nsdk.subtract(%d, %d) = %d' % (a, b, result))
        self.assertEqual(result, 16)

    def test_httpGet(self, ip='203.208.48.146'):
        result = self.sdk.httpGet(ip)
        print('\nsdk.httpGet(%s) is %s' % (ip, result))
        self.assertIsNotNone(result)

    def test_getDeviceInfo(self):
        result = self.sdk.getDeviceInfo()
        print('\nsdk.getDeviceInfo() is:\n%s' % (result))
        self.assertIsNotNone(result)
    


if __name__ == '__main__':
    unittest.main(verbosity=2)

File: RPC-TestMySDK/src/com/mrqyoung/rpc/testmysdk/TestMySDK.java
package com.mrqyoung.rpc.testmysdk;


import java.net.URL;

import junit.framework.Assert;
import junit.framework.TestCase;

import org.apache.xmlrpc.XmlRpcException;
import org.apache.xmlrpc.client.XmlRpcClient;
import org.apache.xmlrpc.client.XmlRpcClientConfigImpl;
import org.junit.After;
import org.junit.Before;

public class TestMySDK extends TestCase {
	
	private static final String RPC_SERVER_URI = "http://192.168.1.4:8000/";
	private static XmlRpcClient rpc;

	@Before
	public void setUp() throws Exception {
		XmlRpcClientConfigImpl config = new XmlRpcClientConfigImpl();
		config.setServerURL(new URL(RPC_SERVER_URI));
		rpc = new XmlRpcClient();
		rpc.setConfig(config);
	}

	@After
	public void tearDown() throws Exception {
	}
	
	public void testAdd() throws XmlRpcException {
		Object[] params = new Object[]{123, 456};
		int result = (Integer) rpc.execute("sdk.add", params);
		System.out.println("sdk.add(123, 456) = " + result);
		Assert.assertEquals(579, result);
	}
	
	public void testSubtract() throws XmlRpcException {
		Object[] params = new Object[]{2015, 1999};
		int result = (Integer) rpc.execute("sdk.subtract", params);
		System.out.println("sdk.subtract(2015, 1999) = " + result);
		Assert.assertEquals(16, result);
	}
	
	public void testHttpGet() throws XmlRpcException {
		Object[] params = new Object[]{"203.208.48.146"};
		String result = (String) rpc.execute("sdk.httpGet", params);
		System.out.println("sdk.httpGet(203.208.48.146) is " + result);
		Assert.assertNotNull(result);
	}
	
	public void testGetDeviceInfo() throws XmlRpcException {
		String result = (String) rpc.execute("sdk.getDeviceInfo", new Object[]{});
		System.out.println("sdk.getDeviceInfo() is:\n" + result);
		Assert.assertNotNull(result);
	}

}
